File: python/perfect-numbers/perfect_numbers.py
def classify(number):
    aliquot_sum = 0
    i = 1
    if number == 0 or number < 0:
        raise ValueError("The number cannot be 0 or negative!")

    while i < number:
        if number % i == 0:
            aliquot_sum += i
        i += 1
    if aliquot_sum == number:
        return "perfect"
    elif aliquot_sum > number:
        return "abundant"
    else:
        return "deficient"

File: csharp/perfect-numbers/PerfectNumbers.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
        var aliquotSum = GetFactors(number).Sum();
        
        if (number < 1)
        {
            throw new ArgumentOutOfRangeException("not a natural number");
        }
        
        if (aliquotSum == number)
        {
            return Classification.Perfect;
        } else if (aliquotSum > number)
        {
            return Classification.Abundant;
        } else
        {
            return Classification.Deficient;
        }
    }

    private static IEnumerable<int> GetFactors(int number)
    {
        var factors = new HashSet<int>();

        for (int i = 1; i < number; i++)
        {
            if (number % i == 0)
            {
                factors.Add(i);
            }
        }
        return factors;
    }
}
